template/macros.jinja

```diff
@@ -43,22 +42,29 @@
             - salt://xxx/files/default/etc/yyy/zzz.conf.jinja
           - template: jinja
   #}
-  {%- set path_prefix = prefix|replace(':', '/') %}
-  {%- set files_switch_list = salt['pillar.get'](prefix ~ ':files_switch',
-                                           default_files_switch) %}
-  {%- for grain in files_switch_list if grains.get(grain) is defined %}
+  {%- set formula = 'template' %}
+  {%- set path_prefix = salt['config.get'](formula ~ ':tofs:path_prefix', formula) %}
+  {%- set files_dir = salt['config.get'](formula ~ ':tofs:dirs:files', 'files') %}
+  {%- set files_switch_list = salt['config.get'](
+      formula ~ ':files_switch',
+      default_files_switch
+  ) %}
+  {#- Append an empty value to evaluate as `default` in the loop below #}
+  {%- do files_switch_list.append('') %}
+  {%- for fs in files_switch_list %}
     {%- for file in files %}
-      {%- set url = '- salt://' ~ '/'.join([path_prefix,
-                                            'files',
-                                            grains.get(grain),
-                                            file.lstrip('/')]) %}
+      {%- if fs %}
+        {%- set fs_dir = salt['config.get'](fs, fs) %}
```

Comment:
If the "grain" isn't found, it uses it as a literal value in the path instead.  This has two noticeable benefits:

1. Allows for additional path matching from the pillar.
1. Easier to debug when not working, since the unmatched value will be shown directly in the URL.